test/e2e/lifecycle/deployment_test.go

```diff
@@ -0,0 +1,65 @@
+package test
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+
+	. "github.com/kubevirt/cluster-network-addons-operator/test/check"
+	. "github.com/kubevirt/cluster-network-addons-operator/test/operations"
+	. "github.com/kubevirt/cluster-network-addons-operator/test/releases"
+)
+
+var _ = Context("Cluster Network Addons Operator", func() {
+	Context("when installed from master release", func() {
+		masterRelease := LatestRelease()
+		BeforeEach(func() {
+			InstallRelease(masterRelease)
+		})
+
+		It("successfully turns ready", func() {
+			CheckOperatorIsReady(10 * time.Minute)
+		})
+
+		Context("and when NodeNetworkConfig with supported spec is created", func() {
+			BeforeEach(func() {
+				CreateConfig(masterRelease.SupportedSpec)
+			})
+
+			It("reaches Available condition with all containers using expected images", func() {
+				CheckConfigCondition(ConditionAvailable, ConditionTrue, 10*time.Minute, CheckDoNotRepeat)
+				CheckReleaseUsesExpectedContainerImages(masterRelease)
+			})
+
+			It("stays in Available condition while the operator is being removed and redeployed", func() {
+				configIsAvailable := func() {
+					CheckConfigCondition(ConditionAvailable, ConditionTrue, CheckImmediately, CheckDoNotRepeat)
+				}
+
+				reinstallingOperator := func() {
```

Comment:
Isn't this different test ?

Reply:
What do you mean? This is a test reinstalling operator and making sure it doesn't affect operand.

Reply:
Smell like context needed "when operator is re-installed" but maybe this is too ginkgo idiomatic stuff.